tests/rbac/test_middleware.py

```diff
@@ -273,6 +274,74 @@ def test_tenant_process_without_org_id(self):
         self.assertEqual(Tenant.objects.filter(tenant_name="test_user").count(), 1)
         self.assertEqual(Tenant.objects.filter(tenant_name="test_user").first().org_id, None)
 
+    def test_should_load_user_permissions_org_admin(self):
+        """Tests that the function that determines if user permissions should be loaded returns False for org admins."""
+        user = User()
+        user.admin = True
+
+        middleware = IdentityHeaderMiddleware(get_response=Mock())
+        self.assertEqual(middleware.should_load_user_permissions(Mock(), user), False)
+
+    def test_should_load_user_permissions_regular_user_non_access_endpoint(self):
+        """Tests that the function under test returns True for regular users who have requested a path which isn't the access path"""
+        user = User()
+        user.admin = False
+
+        request = Mock()
+        request.path = "/principals/"
+
+        middleware = IdentityHeaderMiddleware(get_response=Mock())
+        self.assertEqual(middleware.should_load_user_permissions(request, user), True)
+
+    def test_should_load_user_permissions_regular_user_access_non_get_request(self):
+        """Tests that the function under test returns True for regular users who have requested the access path but with a different HTTP verb than GET"""
+        user = User()
+        user.admin = False
+
+        request = Mock()
+        request.path = "/access/"
+
+        middleware = IdentityHeaderMiddleware(get_response=Mock())
+
+        invalid_verbs = ["DELETE", "POST", "PATCH"]
+        for verb in invalid_verbs:
+            request.method = verb
+            self.assertEqual(middleware.should_load_user_permissions(request, user), True)
+
+    def test_should_load_user_permissions_regular_user_access(self):
+        """Tests that the function under test returns True for regular users who have requested the access path with the expected query parameters"""
+        user = User()
+        user.admin = False
+
+        request = Mock()
+        request.path = "/access/"
+        request.method = "GET"
+        request.GET = QueryDict("application=rbac&username=foo")
+        middleware = IdentityHeaderMiddleware(get_response=Mock())
+        self.assertEqual(middleware.should_load_user_permissions(request, user), True)
+
+    def test_should_load_user_permissions_regular_user_access_missing_query_params(self):
```

Comment:
the `application` param is mandatory so it is enforced in the` rbac/rbac/midleware.py `

Reply:
I don't follow on this one, what do you mean?

Reply:
this was related to my first comment about the `application` param, you left the param in the condition so this comment is no longer valid